pennylane/labs/resource_estimation/ops/op_math/controlled_ops.py

```diff
@@ -0,0 +1,1687 @@
+
+        Resources:
+            The resources are expressed as one general :class:`~.ResourceMultiControlledX` gate.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+                represents a specific quantum gate and the number of times it appears
+                in the decomposition.
+        """
+        mcx = resource_rep(
+            re.ResourceMultiControlledX,
+            {
+                "num_ctrl_wires": ctrl_num_ctrl_wires + 2,
+                "num_ctrl_values": ctrl_num_ctrl_values,
+            },
+        )
+        return [GateCount(mcx)]
+
+    @classmethod
+    def default_pow_resource_decomp(cls, pow_z, elbow=None) -> list[GateCount]:
+        r"""Returns a list representing the resources for an operator raised to a power.
+
+        Args:
+            pow_z (int): the power that the operator is being raised to
+
+        Resources:
+            This operation is self-inverse, thus when raised to even integer powers acts like
+            the identity operator and raised to odd powers it produces itself.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+                represents a specific quantum gate and the number of times it appears
+                in the decomposition.
+        """
+        return (
+            [GateCount(resource_rep(re.ResourceIdentity))]
+            if pow_z % 2 == 0
+            else [GateCount(cls.resource_rep(elbow=elbow))]
+        )
+
+
+class ResourceMultiControlledX(ResourceOperator):
+    r"""Resource class for the MultiControlledX gate.
+
+    Args:
+        wires (Union[Wires, Sequence[int], or int]): control wire(s) followed by a single target wire (the last entry of ``wires``) where
+            the operation acts on
+        control_values (Union[bool, list[bool], int, list[int]]): The value(s) the control wire(s)
+            should take. Integers other than 0 or 1 will be treated as ``int(bool(x))``.
+
+    Resource Parameters:
+        * num_ctrl_wires (int): the number of qubits the operation is controlled on
+        * num_ctrl_values (int): the number of control qubits, that are controlled when in the :math:`|0\rangle` state
+
+    Resources:
+        The resources are obtained from Table 3 of `Claudon, B., Zylberman, J., Feniou, C. et al.
+        <https://www.nature.com/articles/s41467-024-50065-x>`_. Specifically, the
+        resources are defined as the following rules:
+
+        * If there is only one control qubit, treat the resources as a :class:`~.ResourceCNOT` gate.
+
+        * If there are two control qubits, treat the resources as a :class:`~.ResourceToffoli` gate.
+
+        * If there are three control qubits, the resources are two :class:`~.ResourceCNOT` gates and one :class:`~.ResourceToffoli` gate.
+
+        * If there are more than three control qubits (:math:`n`), the resources are defined as :math:`36n - 111` :class:`~.ResourceCNOT` gates.
```

Comment:
We should remove this from here because actually it will depend on what the default is. Specifically I think we should set the unary iterator decomposition as the default and then have the one referenced here as builtin one.